### ISSUE TITLE
Insights: sort capture group series order on live preview 

### ISSUE THREAD

**Issue by Joelkw** [sourcegraph/sourcegraph#34492]
We made it so that legends sort correctly on display in #29949: 
![image](https://user-images.githubusercontent.com/11967660/165234260-eba6d52d-868b-4432-952d-f5fd749e9c2c.png)

But not on live preview: 
![Screen Shot 2022-04-25 at 11 15 11 PM](https://user-images.githubusercontent.com/11967660/165234287-562c06d1-eb6c-44c7-be3d-c11e8599fa18.png)

We should match the sort order across both types.

**Comment by sourcegraph-bot-2** [comment]
Heads up @joelkw @felixfbecker @vovakulikov @unclejustin - the "team/code-insights" label was applied to this issue.

**Comment by Joelkw** [comment]
@felixfbecker + frontend folks – do you think there is a larger effort we need here to unify the live preview component? So we can get work like this for free in the fugure?

**Comment by vovakulikov** [comment]
@Joelkw I don't think this is actually possible at the moment while we don't have control over series orders on the backend. The thing that happens here is `searchPreview` GQL handler returns different series order compare to insightViews series GQL handler (the one that we use on the dashboard page). I think this problem could be solve after we introduce order filtes for series on the backend) cc @coury-clark

**Comment by coury-clark** [comment]
It's true that this should be resolved when we implement explicit sorting options; however, if this is a priority I don't see any reason we shouldn't be able to sort these series on the frontend in the same way. It's the same API response, so there isn't any clear reason to me why one insight type would be sortable and one isn't.

**Comment by vovakulikov** [comment]
@coury-clark Agree. This is actually a good example of where we should be more careful with our API (the track that we touched on the last retro).

**Comment by vovakulikov** [comment]
@sourcegraph/code-insights-backend since that the sort logic was removed from the frontend (because now it lives on the backend) I think we should make this change on the backend and sort series there instead of having sem ver sorting on the frontend

My concerns about FE sorting for the live preview 
- We again increase inconsistency between live preview and dashboard insight (dahsobard has insights with already sorted series, live preview still has to sort it on the frontend) 
- Increase inconsistency between FE and BE. If we change sorting for the dashboard insights we need to update it on the live preview on the frontend (in case if we do sorting on the frotnend for the live preview) 
- Last time sort logic increased our bundle on 30kb (I belive because we used big lib for it and we could come up with better solution there)

**Comment by leonore** [comment]
@vovakulikov we're happy doing this on the backend but this won't get in 4.0

**Comment by vovakulikov** [comment]
@leonore, I understand, there is no need to include this in the 4.0 release. Thanks for checking.
